src/query/src/datafusion.rs

```diff
@@ -373,6 +380,73 @@ impl QueryExecutor for DatafusionQueryEngine {
     }
 }
 
+fn convert_filter_to_df_filter(filter: Expr) -> Result<DfExpr> {
+    match filter {
+        Expr::BinaryOp { left, op, right } => {
+            let left = convert_filter_to_df_filter(*left).unwrap();
+            let right = convert_filter_to_df_filter(*right).unwrap();
+            match op {
+                BinaryOperator::Eq => Ok(left.eq(right)),
+                _ => {
+                    todo!()
```

Comment:
:+1: Maybe returning an unimplemented error is better.

Reply:
Done